### PR TITLE
apps: improve text-editor

### PR DESCRIPTION
**Pull request by feliwir**
- [x] Add loading callback
- [x] Add save callback
- [ ] Add new callback
- [ ] Add multiple tabs

### CODE REVIEW

File: userspace/libraries/libwidget/model/TextModel.h [hunk 1]
```diff
@@ -128,7 +128,7 @@ class TextModel :
 public:
     static RefPtr<TextModel> empty();
 
-    static RefPtr<TextModel> from_file(const char *path);
+    static RefPtr<TextModel> from_file(String path);
```
Comment (sleepy-monax) [inline userspace/libraries/libwidget/model/TextModel.h:131]: This function should be renamed load to match save

File: userspace/libraries/libfilepicker/model/DirectoryListing.h [hunk 2]
```diff
@@ -14,7 +14,7 @@ class DirectoryListing : public Widget::TableModel
 {
```
Comment (sleepy-monax) [inline userspace/libraries/libfilepicker/model/DirectoryListing.h:14]: Directory listing and FileListing can become a single class like FileSystemModel if we use a filter Callback 🤔

**Review by sleepy-monax — COMMENTED** [review]
Nice work as always 👍

File: userspace/libraries/libfilepicker/model/DirectoryListing.h (hunk 2 above)
Comment (feliwir) [reply userspace/libraries/libfilepicker/model/DirectoryListing.h:14]: Ah yep, then it would definetly feel less duplicate. Will fix this

**Comment by sleepy-monax** [comment]
Looks good :+1:

**Pull request merged**